metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetector.java

```diff
@@ -112,14 +125,39 @@ void process(AlertTask alertTask) {
         Instant.ofEpochMilli(alertTask.getCurrentExecutionTime()));
 
     if (violationCondition.hasStaticThresholdCondition()) {
-      evaluateForStaticThreshold(
-          iterator,
-          violationCondition,
-          metricAnomalyEventCondition.getMetricSelection().getMetricAttribute());
+      EvaluationResult evaluationResult =
+          evaluateForStaticThreshold(
+              iterator,
+              violationCondition,
+              metricAnomalyEventCondition.getMetricSelection().getMetricAttribute());
+      if (evaluationResult.isViolation()) {
+
+        MetricAnomalyNotificationEvent metricAnomalyNotificationEvent =
+            MetricAnomalyNotificationEvent.newBuilder()
+                .setViolationTimestamp(alertTask.getCurrentExecutionTime())
+                .setChannelId(alertTask.getChannelId())
+                .setEventConditionId(alertTask.getEventConditionId())
+                .setEventConditionType(alertTask.getEventConditionType())
+                .build();
+        EventRecord eventRecord =
+            EventRecord.newBuilder()
+                .setEventType(METRIC_ANOMALY_NOTIFICATION_EVENT_TYPE)
+                .setEventRecordMetadata(Map.of())
+                .setEventValue(metricAnomalyNotificationEvent.toByteBuffer())
+                .build();
+        NotificationEvent notificationEvent =
+            NotificationEvent.newBuilder()
+                .setTenantId(alertTask.getTenantId())
+                .setNotificationEventMetadata(Map.of())
+                .setEventTimeMillis(System.currentTimeMillis())
+                .setEventRecord(eventRecord)
+                .build();
+        eventProducer.publish(notificationEvent);
```

Comment:
As a follow-up, we need enough logging too - if the producer fails, and metrics.